### PR TITLE
explorer: add error message to copy component

### PR DESCRIPTION
**Pull request by oJshua**
#### Problem
Copy to clipboard functionality and restrictions vary between browsers causing it fail sometimes

#### Summary of Changes
Adds error message for when a copy fails. This is due to a lack of permissions. 

Fixes https://github.com/solana-labs/solana/issues/11932
Issue: https://sentry.io/organizations/solana/issues/1843555677

### CODE REVIEW

**Comment by oJshua** [comment]
I wasn't able to find a way to trigger the "copy" permissions dialog at the top of the browser.

**Comment by jstarry** [comment]
Thanks for kicking this off @oJshua! I started playing with the styles and got far enough along that I thought it would be quicker to append onto your PR. Hope you don't mind!

Desktop:
Screen Shot 2021-02-19 at 10 56 34 AM

Mobile:
Screen Shot 2021-02-19 at 10 56 25 AM

File: explorer/src/components/common/Copyable.tsx [hunk 1]
```diff
@@ -1,56 +1,98 @@
...
-  right?: boolean;
-};
+  replaceText?: boolean;
+}) {
+  const [state, setState] = useState<CopyState>("copy");
 
-type State = "hide" | "copy" | "copied";
+  const handleClick = async () => {
+    try {
+      await navigator.clipboard.writeText(text);
```
Comment (jstarry) [inline explorer/src/components/common/Copyable.tsx:18]: note that this is now inside a try catch, your code wasn't catching the thrown error

File: explorer/src/components/common/Copyable.tsx [hunk 2]
```diff
@@ -1,56 +1,98 @@
...
+      setState("copied");
+    } catch (err) {
+      setState("errored");
+    }
+    setTimeout(() => setState("copy"), 1000);
```
Comment (jstarry) [inline explorer/src/components/common/Copyable.tsx:23]: note that this resets the copy state for errors as well now

File: explorer/src/components/common/Copyable.tsx [hunk 3]
```diff
@@ -1,56 +1,98 @@
...
+          title="Please check your browser's copy permissions."
+        ></span>
+      );
+    }
+    return null;
+  }
 
-export function Copyable({ bottom, right, text, children }: CopyableProps) {
-  const [state, setState] = useState<State>("hide");
+  let message = "";
```
Comment (jstarry) [inline explorer/src/components/common/Copyable.tsx:44]: added a message since the span.title message wasn't too discoverable

**Comment by jstarry** [comment]
> I wasn't able to find a way to trigger the "copy" permissions dialog at the top of the browser.

You can trigger with a non-https non-localhost url

**Comment by oJshua** [comment]
Thanks @jstarry !

File: explorer/src/components/common/Copyable.tsx (hunk 1 above)
Comment (oJshua) [reply explorer/src/components/common/Copyable.tsx:18]: Oops, looks like I forgot something.. :)

Comment (oJshua) [reply explorer/src/components/common/Copyable.tsx:18]: I think I missed a commit, but you covered it.

**Pull request merged**